### ISSUE TITLE
Split out collision detection from Sprite

### ISSUE THREAD

**Issue by straker** [straker/kontra#120]
To help reduce file size, we should remove the [`collidesWith`](https://github.com/straker/kontra/blob/master/src/sprite.js#L450-L468) function from the `Sprite` class and add it to its own file and export it.

```js
// collision.js
export function spriteCollidesWith(sprite, object) {
  // collision code from Sprite#collidesWith
}
```

Please create the PR against the `v7` branch as this would be a breaking change.

**Comment by jmlavoier** [comment]
Are thinking about creating it as a helper function, exporting from another file or create a new abstraction using a class Collision?

**Comment by straker** [comment]
Closed via #123
